File: matiasmasso/Mat.NET/DTO/DTOBaseGuidNumFch.cs
using System;

namespace DTO
{
    public class DTOBaseGuidNumFch : DTOBaseGuid
    {
        public int Num { get; set; }
        public DateTime Fch { get; set; }

        public DTOBaseGuidNumFch() : base()
        {
        }

        public DTOBaseGuidNumFch(Guid oGuid) : base(oGuid)
        {
        }

        public static string Formatted(DTOBaseGuidNumFch value)
        {
            string retval = VbUtilities.Format(value.Fch.Year, "0000") + "." + VbUtilities.Format(value.Num, "00000");
            return retval;
        }
    }
}

File: matiasmasso/Mat.NET/DT2/DTOBaseGuidNumFch.vb
Public Class DTOBaseGuidNumFch
    Inherits DTOBaseGuid

    Property num As Integer
    Property fch As Date

    Public Sub New()
        MyBase.New()
    End Sub

    Public Sub New(oGuid As Guid)
        MyBase.New(oGuid)
    End Sub

    Shared Function Formatted(value As DTOBaseGuidNumFch) As String
        Dim retval As String = Format(value.Fch.Year, "0000") & "." & Format(value.Num, "00000")
        Return retval
    End Function
End Class
